src/igtcloud/client/tools/cli.py

```diff
@@ -170,9 +174,69 @@ def version():
     print(__version__)
 
 
+@click.command(short_help="Upgrade this tool to a new version")
+@click.argument('package_version', required=False, default=None)
+def upgrade(package_version):
+    from igtcloud.client import __version__, __source__
+    import requests
+    if package_version is None:
+        resp = requests.get(f'{__source__}/releases/latest', allow_redirects=False)
+        if resp.status_code == 302:
+            redirect_url = resp.headers['Location']
+            tag_prefix = '/tag/'
+            ix = redirect_url.index(tag_prefix)
+            if ix > 0:
+                package_version = redirect_url[ix + len(tag_prefix):]
+                if __version__ == package_version[1:]:
+                    print(f"Already at latest version: {package_version}")
+                    return
+    if package_version is None:
+        print("Cannot determine version")
+        return
+
+    import sys
+    import subprocess
+    if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
```

Comment:
There's probably a reason so, what's the reason you do not do `if not sys.executable` or something like that ?

Reply:
See: https://pyinstaller.org/en/stable/runtime-information.html